org.eclipse.buildship.ui/src/main/java/org/eclipse/buildship/ui/launch/JavaHomeTab.java

```diff
@@ -130,9 +130,9 @@ public boolean isValid(ILaunchConfiguration launchConfig) {
             return false;
         }
 
-        File javaHome = FileUtils.getAbsoluteFile(javaHomeResolved).orNull();
+        File javaHome = FileUtils.getAbsoluteFile(javaHomeResolved).orElse(null);
```

Comment:
This shows that we shouldn't be using `Optional` here, because we just want null anyway ;) We should just return null when there is no error. 

Optional is only meant for cases where returning null has big potential for confusion, e.g. `stream.findFirst` (where null could mean "the first element is null" or "there is no first element").